### ISSUE TITLE
[a11y] Add option to always allow right-click 

### ISSUE THREAD

**Issue by crogonint** [brave/brave-browser#25715]
## Description 
Request to provide an accessibility feature to always allow the right click menu. As we know, some websites disable the use of the right-click menu. some for good reasons, some just to be irritating. Brave just blocked the extension that I was using, which was the most popular one, due to it containing malware. Since there is NOT a chromium accessibility option to always allow it, people who rely on various extensions to magnify the screen or use a special feature to input characters (or others) can't access them on the right-click menu.

I use Brave BECAUSE Chrome is a google-tastic train-wreck of anti-privacy features. Forcing end-users to rely on (Google's Chrome web store for..) third-party extensions for insanely simple features like "Always Allow Right-Click" is ridiculous (in my humble opinion). 

I would note that there are probably a half dozen other simple features that Google has killed off to attempt to maintain.. whatever their anti-privacy practices dictate. I know one that irritates me personally is auto-blocking insecure HTML connections ( http://____ ). Generally a good idea, yes, but irritates the crap out of me when connecting to routers and switches.. which I do a lot, since I'm a network technician. 

## Expected result:
Give end-users a reliable option for what used to be standard features, so that they don't have to run off and trust questionable extensions from the Chrome web store.

## Brave version (info found on brave://version)
[Version 1.44.101 Chromium: 106.0.5249.65 (Official Build) (64-bit)](https://brave.com/latest/)

**Comment by rebron** [comment]
@crogonint What was the extension?

**Comment by sangwoo108** [comment]
This would be a good start point
//third_party/blink/renderer/core/dom/events/event_dispatcher.cc
```diff
@ -405,8 +413,18 @@ inline void EventDispatcher::DispatchEventPostProcess(
   // Call default event handlers. While the DOM does have a concept of
   // preventing default handling, the detail of which handlers are called is an
   // internal implementation detail and not part of the DOM.
-  if (!event_->defaultPrevented() && !event_->DefaultHandled() &&
-      is_trusted_or_click) {
+
+  // Check if contextmenu event should bypass preventDefault (when Shift key is
+  // pressed)
+  bool bypass_context_menu_prevent_default = false;
+  if (event_->type() == event_type_names::kContextmenu) {
+    if (auto* context_mouse_event = DynamicTo<MouseEvent>(event_)) {
+      bypass_context_menu_prevent_default = context_mouse_event->shiftKey();
+    }
+  }
+
+  if ((!event_->defaultPrevented() || bypass_context_menu_prevent_default) &&
+      !event_->DefaultHandled() && is_trusted_or_click) {
     // Non-bubbling events call only one default event handler, the one for the
     // target.
     node_->DefaultEventHandler(*event_);
```

**Comment by crogonint** [comment]
> [@crogonint](https://github.com/crogonint) What was the extension?

I am SO so sorry! I missed your question.. three years ago when you asked. I can't recall now. If I remember it was something butt-simple, like "Always-Allow-Right-Click", but I wouldn't swear to that. :/

I see the features about to roll out! That's awesome news, I will have to switch back to using Brave as my main browser again. :D

**Comment by sangwoo108** [comment]
> > [@crogonint](https://github.com/crogonint) What was the extension?
> 
> I am SO so sorry! I missed your question.. three years ago when you asked. I can't recall now. If I remember it was something butt-simple, like "Always-Allow-Right-Click", but I wouldn't swear to that. :/
> 
> I see the features about to roll out! That's awesome news, I will have to switch back to using Brave as my main browser again. :D

Sorry for keeping you wait 🙏 Please let us know if it doesn't work for some sites. This is the very basic implmentatiom so I'm wondering it could work in the wild.

**Comment by crogonint** [comment]
I haven't had it fail yet, but I'm busy shutting down.. 837 more tabs in Chrome, saving bookmarks, and getting ready to switch back to Brave permanently. In other words, it might be a few weeks before I get feedback out of Brave on all of my regular websites. For example, I still have GitHub running in Chrome, but I'll log out here and switch it to Brave shortly. :)

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.87.166 Chromium: 144.0.7559.59 (Official Build) beta (64-bit)
-- | --
Revision | 5f2627224f3d5912710a3766aaa1c66426d7321a
OS | Windows 11 Version 25H2 (Build 26200.7462)
```

Verified using the testplan from https://github.com/brave/brave-core/pull/31770#issue-3506984667


1. Installed 1.87.166
2. launch Brave
3. opened brave://flags in a new tab
4. enabled `#force-context-menu-on-shift-right-click`
5. opened following in a new tab
- https://developer.mozilla.org/en-US/play?uuid=475366d5-c058-4f7c-8ca9-b74e95e9c5a2&state=dY%2BxbsMwDER%2FheDkAIa7O7aHtBkKtFtGLYp1jdxalGBRRYsg%2F144DbIUXUjicI%2B4O7PXMHPLXaLJ9YYlPkZRfOkrpBgeDh40%2FioUIIW8zXQEhNyU7XGGoyikfsqU7GJPi02%2B6R7SYKRLw64oTXplJOo%2FXBTcCK55zJlb5prf1z1GyUr3TNSTi2MJEG1O0P2M9dx9P7vqT%2FLN1oiRu9hY5%2FafEH2ZskKwVIZvvcLVX1OFDfUDnY0QoUkLVvsT3myZtVrfXdbBNatHALfs7PLBlx8%3D&srcPrefix=%2Fen-US%2Fdocs%2FWeb%2FAPI%2FElement%2Fcontextmenu_event%2F
6. clicked "Run" button for the script on the left
7. clicking the first paragraph with right click without any keys shouldn't open context menu
8. clicking the first paragraph with right click and shift key should open context menu

https://github.com/user-attachments/assets/91a23bcd-6845-49da-b0b3-43fc56d8c936
